x/dex/keeper/core_helper.go

```diff
@@ -521,3 +526,61 @@ func (k Keeper) BurnShares(ctx sdk.Context, addr sdk.AccAddress, amount sdk.Int,
 	}
 	return nil
 }
+
+///////////////////////////////////////////////////////////////////////////////
+//                            CORE VALIDATION                                //
+///////////////////////////////////////////////////////////////////////////////
+
+func (k Keeper) ValidateDeposit(
+	ctx sdk.Context,
+	callerAddr sdk.AccAddress,
+	feeTiers []types.FeeTier,
+	feeIndex uint64,
+	pair *types.TradingPair,
+	tickIndex int64,
+	amount0 sdk.Int,
+	amount1 sdk.Int,
+	balance0 sdk.Int,
+	balance1 sdk.Int,
+) (int64, int64, int64, error) {
+	// check that feeIndex is a valid index of the fee tier
```

Comment:
Do we have a test case for each of these checks?

Reply:
Added